src/subclass/widget.rs

```diff
@@ -0,0 +1,249 @@
+                .expect("No parent class impl for \"adjust_size_request\"");
+            f(
+                widget.to_glib_none().0,
+                orientation.to_glib(),
+                minimum_size as *mut i32,
+                natural_size as *mut i32,
+            )
+        }
+    }
+
+    fn parent_button_press_event(&self, widget: &Widget, event: &mut gdk::EventButton) -> bool {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            let f = (*parent_class)
+                .button_press_event
+                .expect("No parent class impl for \"button_press_event\"");
+            f(widget.to_glib_none().0, event.to_glib_none_mut().0) != 0
+        }
+    }
+
+    fn parent_button_release_event(&self, widget: &Widget, event: &mut gdk::EventButton) -> bool {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            let f = (*parent_class)
+                .button_release_event
+                .expect("No parent class impl for \"button_release_event\"");
+            f(widget.to_glib_none().0, event.to_glib_none_mut().0) != 0
+        }
+    }
+}
+
+unsafe impl<T: ObjectSubclass + WidgetImpl> IsSubclassable<T> for WidgetClass {
+    fn override_vfuncs(&mut self) {
+        unsafe extern "C" fn widget_adjust_baseline_allocation<T: ObjectSubclass>(
+            ptr: *mut gtk_sys::GtkWidget,
+            baseptr: *mut i32,
+        ) where
+            T: WidgetImpl,
+        {
+            let instance = &*(ptr as *mut T::Instance);
+            let imp = instance.get_impl();
+            let wrap: Widget = from_glib_borrow(ptr);
+
+            imp.adjust_baseline_allocation(&wrap, &mut *baseptr)
+        }
+
+        unsafe extern "C" fn widget_adjust_baseline_request<T: ObjectSubclass>(
+            ptr: *mut gtk_sys::GtkWidget,
+            minptr: *mut i32,
+            natptr: *mut i32,
+        ) where
+            T: WidgetImpl,
+        {
+            let instance = &*(ptr as *mut T::Instance);
+            let imp = instance.get_impl();
+            let wrap: Widget = from_glib_borrow(ptr);
+
+            imp.adjust_baseline_request(&wrap, &mut *minptr, &mut *natptr)
+        }
+
+        unsafe extern "C" fn widget_adjust_size_allocation<T: ObjectSubclass>(
+            ptr: *mut gtk_sys::GtkWidget,
+            orientation: gtk_sys::GtkOrientation,
+            minptr: *mut i32,
+            natptr: *mut i32,
+            posptr: *mut i32,
+            sizeptr: *mut i32,
+        ) where
+            T: WidgetImpl,
+        {
+            let instance = &*(ptr as *mut T::Instance);
+            let imp = instance.get_impl();
+            let wrap: Widget = from_glib_borrow(ptr);
+
+            let wrap_orientation: Orientation = match orientation {
```

Comment:
Why not `from_glib(orientation)`?

Reply:
`gtk::Orientation` doesn't implement any of the `FromGlib*` traits

Reply:
It does: https://github.com/gtk-rs/gtk/blob/4e3d79cecc70013f5be1b9ff419b60daaf81ec55/src/auto/enums.rs#L3333-L3356

Reply:
Ah, I missed that when reviewing the docs. Thanks.